src/create-glamorous.js

```diff
@@ -149,6 +149,15 @@ function createGlamorous(splitProps) {
         },
       })
 
+      function withComponent(newComp) {
+        return Object.assign(GlamorousComponent, {
```

Comment:
Ah yes! We don't want to be manipulating the existing component, we want to make a _new_ one that has the same properties! So instead this function should probably be:

```javascript
function withComponent(newComp, options) {
  return glamorous(newComp, {
    ...options,
    forwardProps: GlamorousComponent.forwardProps,
  })(GlamorousComponent.styles)
}
```

I _think_ that should do it....